### ISSUE TITLE
DNS over TCP(ARES_FLAG_USEVC), wait response timeout when send multi query at once

### ISSUE THREAD

**Issue by fooofei** [c-ares/c-ares#266]
### brief

use DNS over TCP, success of query one name at once,

fail of query multi names at once.

### details

Use `ARES_FLAG_USEVC` to specific only use TCP, like this
```c
opts.flags = ARES_FLAG_USEVC;
optmask = ARES_OPT_FLAGS;
rc = ares_init_options(&ch, &opts, optmask);
```

Then called `ares_gethostbyname` with different query name,

`c-ares` will compose all query to one TCP segment.

Fail wait response of multi querys at one TCP segment,

but spilt up query name, will success of one query at one TCP segment.

### others

Some env success of query multi names at once.

**Comment by fooofei** [comment]
pcap files :
[dns-one-good.pcapng.zip](https://github.com/c-ares/c-ares/files/3380296/dns-one-good.pcapng.zip)
[dns-multi-good.pcapng.zip](https://github.com/c-ares/c-ares/files/3380297/dns-multi-good.pcapng.zip)
[dns-multi-bad.pcapng.zip](https://github.com/c-ares/c-ares/files/3380298/dns-multi-bad.pcapng.zip)

**Comment by fooofei** [comment]
@yangtse  @bagder can you give some help? thx.

**Comment by fooofei** [comment]
adig output
```
$ ./adig -f usevc -d -s 114.114.114.114 x.com y.com
Answer for query x.com:
Timeout while contacting DNS servers
Answer for query y.com:
Timeout while contacting DNS servers
```
x.com is domain1, y.com is domain2

**Comment by bradh352** [comment]
is this a regression from a prior version?  what version of c-ares is being used?

**Comment by fooofei** [comment]
@bradh352  c-ares version is c-ares 1.15.0 , download from https://c-ares.haxx.se/download/c-ares-1.15.0.tar.gz

**Comment by fooofei** [comment]
c-ares-1.13.0  has the same issue.

**Comment by bradh352** [comment]
Fixed in #552
